### DISCUSSION TITLE
Copilot chat unable to view context in Visual Studio project

### DISCUSSION THREAD

**Discussion by DrProply** [community/community#88127] [Copilot Conversations]
### Select Topic Area

Bug

### Body

In specific Visual Studio projects, I am having problems where Copilot chat appears to be unable to view context for the files I have open, even when I explicitly provide this context by highlighting a region of text and triggering Copilot chat via the right click menu. For example, when I ask Copilot chat to /explain the highlighted text, I simply get the following response:

"I apologize, but I cannot provide and explanation for the selected code as you haven't provided any specific code for me to analyze. Could you please provide the code you would like me to explain?"

Automatic Copilot line completion works fine in this project, but Copilot chat is unable to provide any response unless I copy and paste the code into the query window (at which point it presumably interprets this without the larger context of the project). Copilot chat works as expected in some of my other Visual Studio projects. Any ideas for how to fix this?

**Comment by vinaythippeswamy** [comment]
I have a business subscription for co-pilot chat and have a very similar problem. I am using VSCode and have installed the Co-pilot chat plugin and have logged in into github from IDE. Settings on Github.com clearly shows that chat in IDE is enabled for co-pilot. It worked for the first few days and now it does not have access to the code thats open in the editor. It cannot explain the code, nor write test cases and keeps saying it cannot access code and that I need to copy paste the code into the chat window for it to explain.

PS: I have installed Idea and logged in into github from there as well (not the chat , just the co-pilot). Is it possible that logging in from Idea to github co-pilot somehow disables the abilty of Co-pilot chat, to access code open in the editor in VSCode due to some subscription model?

Really appreciate any help. Oh BTW, tried both VSCode and VSCode insiders and no luck.

**Comment by DrProply** [comment]
Updating VS2022 fixed this for me - not sure why it wasn't a problem for some projects before updating, but happy that it is working across the board now.
